client/lib/validation.rs

```diff
@@ -287,23 +287,38 @@ pub(crate) fn validate_get_block_response(
         return Ok(());
     };
     let block = Block::from(json_block);
-    block.verify()?;
-    match maybe_block_identifier {
-        Some(BlockIdentifier::Hash(block_hash)) => {
-            if block_hash != block.hash() {
-                return Err(ValidateResponseError::UnexpectedBlockHash);
-            }
-        }
-        Some(BlockIdentifier::Height(height)) => {
-            // More is necessary here to mitigate a MITM attack
-            if height != &block.height() {
-                return Err(ValidateResponseError::UnexpectedBlockHeight);
+
+    // Currently, the consumer of this function (the client app) has no access to the chainspec,
+    // hence, it cannot give us the proper `merkle_tree_hash_activation` which is needed
+    // for block verification. Therefore, we try to validate the block with both legacy
+    // and new hashing algorithm. If one of the validation passes, we assume the block is correct.
+    // This is a temporary solution which should be removed once the client is given
+    // access to the proper `merkle_tree_hash_activation`.
+    let result_legacy_hash = block.verify(EraId::from(0u64));
+    let result_new_hash = block.verify(block.header().era_id() + 1);
+
+    match (result_legacy_hash, result_new_hash) {
```

Comment:
Would it possible to change to this to a if statement
if `result_legacy_hash.is_ok()` || `result_new_hash.is_ok()`

Reply:
Thanks! Good point.
Fixed here with the "early exit" approach: https://github.com/casper-network/casper-node/pull/2531/commits/0d334a753fe3f3d86216a1d148e8f72325ab84c4